conda_forge_tick/git_utils.py

```diff
@@ -259,6 +310,100 @@ def delete_branch(ctx: GithubContext, pr_json: LazyJson, dry_run: bool = False)
     pr_json["head"]["ref"] = "this_is_not_a_branch"
 
 
+def trim_pr_josn_keys(
+    pr_json: Union[Dict, LazyJson],
+    src_pr_json: Optional[Union[Dict, LazyJson]] = None,
+) -> Union[Dict, LazyJson]:
+    """Trim the set of keys in the PR json. The keys kept are defined by the global
+    PR_KEYS_TO_KEEP.
+
+    Parameters
+    ----------
+    pr_json : dict-like
+        A dict-like object with the current PR information.
+    src_pr_json : dict-like, optional
+        If this object is sent, the values for the trimmed keys are taken
+        from this object. Otherwise `pr_json` is used for the values.
+
+    Returns
+    -------
+    pr_json : dict-like
+        A dict-like object with the current PR information trimmed to the subset of
+        keys.
+    """
+    # keep a subset of keys
+    def _munge_dict(dest, src, keys):
+        for k, v in keys.items():
+            if k in src:
+                if v is None:
+                    dest[k] = src[k]
+                else:
+                    dest[k] = {}
+                    _munge_dict(dest[k], src[k], v)
+
+    if src_pr_json is None:
+        src_pr_json = copy.deepcopy(dict(pr_json))
+
+    pr_json.clear()
+    _munge_dict(pr_json, src_pr_json, PR_KEYS_TO_KEEP)
+    return pr_json
+
+
+def lazy_update_pr_json(
+    pr_json: Union[Dict, LazyJson],
+    ctx: GithubContext,
+    force: bool = False,
+    trim: bool = True,
+) -> Union[Dict, LazyJson]:
+    """Lazily update a GitHub PR.
+
+    This function will use the ETag in the GitHub API to update PR information
+    lazily. It sends the ETag to github properly and if nothing is changed on their
+    end, it simply returns the PR. Otherwise the information is refershed.
+
+    Parameters
+    ----------
+    pr_json : dict-like
+        A dict-like object with the current PR information.
+    ctx : GithubContext
+        The context object with GitHub credential information.
+    force : bool, optional
+        If True, forcibly update the PR json even if it is not out of date
+        according to the ETag. Default is False.
+    trim : bool, optional
+        If True, trim the PR json keys to ones in the global PR_KEYS_TO_KEEP.
+        Default is True.
+
+    Returns
+    -------
+    pr_json : dict-like
+        A dict-like object with the current PR information.
+    """
+    hdrs = {
+        "Authorization": f"token {ctx.github_password}",
+        "Accept": "application/vnd.github.v3+json",
+    }
+    if not force and "ETag" in pr_json:
+        hdrs["If-None-Match"] = pr_json["ETag"]
+
+    r = requests.get(
```

Comment:
Does this handle the fake PRs that we write when we don't issue a PR?

Reply:
This is how the code worked previous too. Those fake PRs fed directly to github3 would raise an error as well.